src/nvim/ex_docmd.c

```diff
@@ -4867,50 +4868,58 @@ static struct {
  * List of names for completion for ":command" with the EXPAND_ flag.
  * Must be alphabetical for completion.
  */
-static struct {
-  int expand;
-  char    *name;
-} command_complete[] =
-{
-  { EXPAND_AUGROUP, "augroup" },
-  { EXPAND_BEHAVE, "behave" },
-  { EXPAND_BUFFERS, "buffer" },
-  { EXPAND_CHECKHEALTH, "checkhealth" },
-  { EXPAND_COLORS, "color" },
-  { EXPAND_COMMANDS, "command" },
-  { EXPAND_COMPILER, "compiler" },
-  { EXPAND_CSCOPE, "cscope" },
-  { EXPAND_USER_DEFINED, "custom" },
-  { EXPAND_USER_LIST, "customlist" },
-  { EXPAND_DIRECTORIES, "dir" },
-  { EXPAND_ENV_VARS, "environment" },
-  { EXPAND_EVENTS, "event" },
-  { EXPAND_EXPRESSION, "expression" },
-  { EXPAND_FILES, "file" },
-  { EXPAND_FILES_IN_PATH, "file_in_path" },
-  { EXPAND_FILETYPE, "filetype" },
-  { EXPAND_FUNCTIONS, "function" },
-  { EXPAND_HELP, "help" },
-  { EXPAND_HIGHLIGHT, "highlight" },
-  { EXPAND_HISTORY, "history" },
+static const char *command_complete[] =
+{
+  [EXPAND_AUGROUP] = "augroup",
+  [EXPAND_BEHAVE] = "behave",
+  [EXPAND_BUFFERS] = "buffer",
+  [EXPAND_CHECKHEALTH] = "checkhealth",
+  [EXPAND_COLORS] = "color",
+  [EXPAND_COMMANDS] = "command",
+  [EXPAND_COMPILER] = "compiler",
+  [EXPAND_CSCOPE] = "cscope",
+  [EXPAND_USER_DEFINED] = "custom",
+  [EXPAND_USER_LIST] = "customlist",
+  [EXPAND_DIRECTORIES] = "dir",
+  [EXPAND_ENV_VARS] = "environment",
+  [EXPAND_EVENTS] = "event",
+  [EXPAND_EXPRESSION] = "expression",
+  [EXPAND_FILES] = "file",
+  [EXPAND_FILES_IN_PATH] = "file_in_path",
+  [EXPAND_FILETYPE] = "filetype",
+  [EXPAND_FUNCTIONS] = "function",
+  [EXPAND_HELP] = "help",
+  [EXPAND_HIGHLIGHT] = "highlight",
+  [EXPAND_HISTORY] = "history",
 #ifdef HAVE_WORKING_LIBINTL
-  { EXPAND_LOCALES, "locale" },
+  [EXPAND_LOCALES] = "locale",
 #endif
-  { EXPAND_MAPPINGS, "mapping" },
-  { EXPAND_MENUS, "menu" },
-  { EXPAND_MESSAGES, "messages" },
-  { EXPAND_OWNSYNTAX, "syntax" },
-  { EXPAND_SYNTIME, "syntime" },
-  { EXPAND_SETTINGS, "option" },
-  { EXPAND_PACKADD, "packadd" },
-  { EXPAND_SHELLCMD, "shellcmd" },
-  { EXPAND_SIGN, "sign" },
-  { EXPAND_TAGS, "tag" },
-  { EXPAND_TAGS_LISTFILES, "tag_listfiles" },
-  { EXPAND_USER, "user" },
-  { EXPAND_USER_VARS, "var" },
-  { 0, NULL }
+  [EXPAND_MAPPINGS] = "mapping",
+  [EXPAND_MENUS] = "menu",
+  [EXPAND_MESSAGES] = "messages",
+  [EXPAND_OWNSYNTAX] = "syntax",
+  [EXPAND_SYNTIME] = "syntime",
+  [EXPAND_SETTINGS] = "option",
+  [EXPAND_PACKADD] = "packadd",
+  [EXPAND_SHELLCMD] = "shellcmd",
+  [EXPAND_SIGN] = "sign",
+  [EXPAND_TAGS] = "tag",
+  [EXPAND_TAGS_LISTFILES] = "tag_listfiles",
+  [EXPAND_USER] = "user",
+  [EXPAND_USER_VARS] = "var",
+  [0] = "",
```

Comment:
is this needed for something? It doesn't fill the role of the `{ 0, NULL }` at least.

Reply:
I previously thought it would be needed if there is some match with 0, but actually the condition was `command_complete[i].expand != 0` and as I modified the code, now it is not needed, I am going to remove it